source/Halibut.Tests/BackwardsCompatibility/TestOldClientWithNewService.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Halibut.Tests.BackwardsCompatibility.Util;
+using Halibut.Tests.Support;
+using Halibut.Tests.Support.BackwardsCompatibility;
+using Halibut.Tests.TestServices;
+using NUnit.Framework;
+
+namespace Halibut.Tests.BackwardsCompatibility
+{
+    public class TestOldClientWithNewService
+    {
+        [Test]
+        public async Task Listening()
+        {
+            new SerilogLoggerBuilder().Build().Information("Hello");
+            var echoService = new CallRecordingEchoServiceDecorator(new EchoService());
+            using (var clientAndService = await PreviousClientVersionAndServiceBuilder.WithListeningService().WithClientVersion(PreviousServiceVersions.v5_0_429)
+                       .WithEchoServiceService(echoService)
+                       .Build())
+            {
+                
+                var echo = clientAndService.CreateClient<IEchoService>();
+                echo.SayHello("hello");
+            }
+
+            echoService.SayHelloCallCount.Should().Be(1);
+        }
+        
+        [Test]
+        public async Task Polling()
+        {
+            new SerilogLoggerBuilder().Build().Information("Hello");
+            var echoService = new CallRecordingEchoServiceDecorator(new EchoService());
+            using (var clientAndService = await PreviousClientVersionAndServiceBuilder.WithPollingService().WithClientVersion(PreviousServiceVersions.v5_0_429)
```

Comment:
Can these 2 tests become 1 with an TestCaseSource to control Polling/Listening so were not repeating test code for all ServiceTypes

Reply:
Will do once https://github.com/OctopusDeploy/Halibut/pull/270 is in. Doing it now will cause a poison merge since websockets are not supported yet.